src/content/docs/browser/new-relic-browser/browser-apis/setuserid.mdx

```diff
@@ -0,0 +1,79 @@
+---
+title: setUserId
+type: apiDoc
+shortDescription: Adds a user-defined identifier string to subsequent events on the page.
+tags:
+  - Browser
+  - Browser monitoring
+  - Browser agent and SPA API
+metaDescription: Browser API call to add an identifier to subsequent events on the page.
+redirects:
+---
+
+<Callout variant="important">
+This API will work for any browser edition (Browser Lite, Pro, or Pro+SPA).
+</Callout>  
+
+## Syntax
+
+```js
+newrelic.setUserId(value: string|null)
+```
+
+Adds a user-defined identifier string to subsequent events on the page.
+
+## Requirements
+
+Agent version [1.230.0](/docs/release-notes/new-relic-browser-release-notes/browser-agent-release-notes) or higher.
+
+## Description
+
+Upon executing this function with a valid value, the browser agent records the value as the `enduser.id` attribute with all events until the attribute is manually unset. The identifier will be stored in the browser, so that subsequent page visits of the _same_ origin **within a session** span retain it on events. Do note that this functionlity may fluctuate depending on end-user browser privacy settings. If this function is called with a `value = null`, any existing user ID will be deleted from **both** the current page's events and the storage.
+
+The ID will be attached to JavaScriptError events in particular for [Errors Inbox](/docs/errors-inbox/errors-inbox/) usage. If you are using [SPA monitoring](/docs/browser/single-page-app-monitoring/get-started/welcome-single-page-app-monitoring) with a compatible agent version, user ID will also be included in [`newrelic.interaction`](/docs/browser/new-relic-browser/browser-agent-apis/browser-spa-api-newrelicinteraction) events.
+
+## Parameters
+
+<table>
+  <thead>
+    <tr>
+      <th width="25%">
+        Parameter
+      </th>
+
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `value`
+
+        _string_ OR _null_
+      </td>
+
+      <td>
+        Required. A string identifier for the end-user, useful for tying all browser events to specific users. The caller is responsible for uniqueness of IDs, if such is desired.
```

Comment:
@cwli24 while this is required, are you saying that the uniqueness isn't? If so, could we revise the sentence like this: "The caller is responsible for the uniqueness of IDs, but `value` doesn't have to be unique."